### ISSUE TITLE
performance: Proposed dynamic icon component contains all icons

### ISSUE THREAD

**Issue by marvin-w** [DanielPantle/strapi-plugin-react-icons#15]
Hey! 

thanks for the plugin! Really cool!

I was looking at the code in the README you proposed:

```ts
import * as ReactIcons from "react-icons/all";

interface IIconComponent {
  icon: string;
  size?: number;
}
type IReactIcon = keyof typeof ReactIcons;

const IconComponent: React.FC<IIconComponent> = ({ icon, size }) => {
  const DynamicIconComponent = ReactIcons[icon as IReactIcon];

  if (undefined === DynamicIconComponent) return <></>;

  return <DynamicIconComponent size={size} />;
};
```

This will include all icons in the final bundle and hence blow it up in size quite significantly. Are you aware of any alternative?

**Comment by DanielPantle** [comment]
Hi, glad you like the plugin :)

I am aware of the problem, but do not know of any alternative until now. Unfortunatelly the problem is even bigger because "react-icons/all" was removed, therefore I had to lock the version to of the react-icons package (see https://github.com/DanielPantle/strapi-plugin-react-icons/issues/5).

If you have suggestions how to fix this they are very welcome!

**Comment by akanshSirohi** [comment]
@marvin-w @DanielPantle Here is the solution: https://github.com/DanielPantle/strapi-plugin-react-icons/issues/22#issuecomment-1793514490

**Comment by DanielPantle** [comment]
Resolved with #26 (v0.3.0)
